File: javascript/hamming/hamming.js
const Hamming = function Hamming() {};

Hamming.prototype.compute = function compute(seqA, seqB) {
  if (seqA.length !== seqB.length) {
    throw Error('DNA strands must be of equal length.');
  }
  return Array.prototype.filter.call(seqA, (e, i) => e !== seqB[i]).length;
};

module.exports = Hamming;

File: go/hamming/hamming.go
// Package hamming provides functions for hamming DNA calculations
package hamming

import "errors"

const testVersion = 6

// Distance calculates the hamming distance between DNA strands a and b
// This is the total number of places that the DNA strands differ. Note that the
// strands must be equal in length
func Distance(a, b string) (int, error) {
	if len(a) != len(b) {
		return -1, errors.New("strings not the same length")
	}
	count := 0
	for i := range a {
		if a[i] != b[i] {
			count++
		}
	}
	return count, nil
}
